Sources/SwiftKuery/ColumnCreator.swift

```diff
@@ -26,4 +26,56 @@ public protocol ColumnCreator {
     /// - Parameter column: The column being built
     /// - Returns: A string representation of the column for the implementing database or nil if it cannot be built
     func buildColumn(for column: Column, using queryBuilder: QueryBuilder) -> String?
+
+    /// Build an appropriate representation of a passed value for the database
+    /// A default implemetation is provided that can be overriden should a
+    /// plugin require non-common behaviour
+    ///
+    /// - Parameter item: The value to convert
+    /// - Returns: A string representing the value that can be passed into the database
+    func packType(_ item: Any, queryBuilder: QueryBuilder) throws -> String
+
+    /// Get the dafault value for a column
+    ///
+    /// - Parameter for: The Column to get the default value for
+    /// - Parameter queryBuilder: The plugin specific queryBuilder
+    /// - Returns: A string representing the default value for the column or nil if no default value is set
+    func getDefaultValue(for column: Column, queryBuilder: QueryBuilder) throws -> String?
+}
+
+public extension ColumnCreator {
+
+    func getDefaultValue(for column: Column, queryBuilder: QueryBuilder) -> String? {
+        guard let defaultValue = column.defaultValue else {
+            if column.nullDefaultValue {
+                return "NULL"
+            }
+            return nil
+        }
+        do {
+            return try packType(defaultValue, queryBuilder: queryBuilder)
+        } catch {
+            return nil
+        }
+    }
+
+    func packType(_ item: Any, queryBuilder: QueryBuilder) throws -> String {
+        switch item {
+        case let val as String:
+            return "'\(val)'"
+        case let val as Bool:
+            return val ? queryBuilder.substitutions[QueryBuilder.QuerySubstitutionNames.booleanTrue.rawValue]
+                : queryBuilder.substitutions[QueryBuilder.QuerySubstitutionNames.booleanFalse.rawValue]
+        case let val as Parameter:
+            return try val.build(queryBuilder: queryBuilder)
+        case let value as Date:
+            if let dateFormatter = queryBuilder.dateFormatter {
+                return dateFormatter.string(from: value)
+            }
+            return "'\(String(describing: value))'"
+        default:
+            let val = String(describing: item)
+            return val == "nil" ? "NULL" : val
```

Comment:
Do we need the `if column.nullDefaultValue` check here?

Reply:
In the current scheme of things, we'll reach here only if a `defaultValue` is present but we aren't able to represent it as a String. I'm not considering scenarios where `packType` will be called from sites outside of `getDefaultValue`.

Reply:
Given we check the passed parameter is not nil prior to calling this function, see `getDefaultValue`, I don't believe we need to return anything other than `String(describing: item)` here.